refactor(PXE)!: move `getNotes` to `debug` submodule

mverzilli opened the pull request:
I would rather not have to explain that `getNotes`, despite being a public method in PXE, is not really intended for production usage in every public facing doc about PXE we write, so I'm proposing to "conceal" it under a `debug` accessor so it looks like:

`pxe.debug.getNotes()` instead of `pxe.getNotes()`

A potential discussion point: moving `getFunctionCall` to `ContractDataProvider`: this function is only used as part of the "invoke contract sync" hack, currently only used by the `getNotes` and `getPrivateEvents`. So it doesn't seem to be a function with a lot of potential for reuse. The decision to move it to `ContractDataProvider` stems from trying to clean PXE a bit and make it available from the new `PXEDebugUtils` module. But I expect this to change and not be necessary once we sort out how we handle contract sync

benesjan commented on docs/docs-developers/docs/resources/migration_notes.md:
```diff
@@ -9,6 +9,16 @@ Aztec is in full-speed development. Literally every version breaks compatibility
 
+### [PXE] deprecated `getNotes`
+
+This function serves debugging purposes so we are taking it out of the main PXE API. If you still need to consume it, you can
```
```suggestion
This function serves only for debugging purposes so we are taking it out of the main PXE API. If you still need to consume it, you can
```

benesjan reviewed (approved):
Makes sense.

mverzilli committed:
refactor(PXE)!: move `getNotes` to `debug` submodule

I would rather not have to explain that `getNotes`, despite being a public method in PXE, is not really intended for production usage in every public facing doc about PXE we write, so I'm proposing to "conceal" it under a `debug` accessor so it looks like:

`pxe.debug.getNotes()` instead of `pxe.getNotes()`

A potential discussion point: moving `getFunctionCall` to `ContractDataProvider`: this function is only used as part of the "invoke contract sync" hack, currently only used by the `getNotes` and `getPrivateEvents`. So it doesn't seem to be a function with a lot of potential for reuse. The decision to move it to `ContractDataProvider` stems from trying to clean PXE a bit and make it available from the new `PXEDebugUtils` module. But I expect this to change and not be necessary once we sort out how we handle contract sync
